include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1680,43 +1766,74 @@ namespace internal
       AssertDimension(fe_coarse.n_dofs_per_cell(), fe_fine.n_dofs_per_cell());
 
 
-      const bool is_feq =
-        fe_fine.n_base_elements() == 1 &&
-        (dynamic_cast<const FE_Q<dim> *>(&fe_fine.base_element(0)) != nullptr);
-
-      // number of dofs on coarse and fine cells
-      transfer.schemes[0].n_dofs_per_cell_coarse =
-        transfer.schemes[0].n_dofs_per_cell_fine =
-          transfer.schemes[1].n_dofs_per_cell_coarse =
-            fe_coarse.n_dofs_per_cell();
-      transfer.schemes[1].n_dofs_per_cell_fine =
-        is_feq ? (fe_fine.n_components() *
-                  Utilities::pow(2 * fe_fine.degree + 1, dim)) :
-                 (fe_coarse.n_dofs_per_cell() *
-                  GeometryInfo<dim>::max_children_per_cell);
-
-      // degree of FE on coarse and fine cell
-      transfer.schemes[0].degree_coarse   = transfer.schemes[0].degree_fine =
-        transfer.schemes[1].degree_coarse = fe_coarse.degree;
-      transfer.schemes[1].degree_fine =
-        is_feq ? (fe_coarse.degree * 2) : (fe_coarse.degree * 2 + 1);
+      const bool is_feq = fe_fine.n_base_elements() == 1 &&
+                          ((dynamic_cast<const FE_Q<dim> *>(
+                              &fe_fine.base_element(0)) != nullptr));
+
+      for (auto &scheme : transfer.schemes)
+        {
+          // number of dofs on coarse and fine cells
+          scheme.n_dofs_per_cell_coarse = fe_coarse.n_dofs_per_cell();
+          scheme.n_dofs_per_cell_fine =
+            is_feq ? (fe_fine.n_components() *
+                      Utilities::pow(2 * fe_fine.degree + 1, dim)) :
+                     (fe_coarse.n_dofs_per_cell() *
+                      GeometryInfo<dim>::max_children_per_cell);
+
+          // degree of FE on coarse and fine cell
+          scheme.degree_coarse = fe_coarse.degree;
+          scheme.degree_fine =
+            is_feq ? (fe_coarse.degree * 2) : (fe_coarse.degree * 2 + 1);
+
+          // reset number of coarse cells
+          scheme.n_coarse_cells = 0;
+        }
+      // correct for first scheme
+      transfer.schemes[0].n_dofs_per_cell_fine = fe_coarse.n_dofs_per_cell();
+      transfer.schemes[0].degree_fine          = fe_coarse.degree;
 
       // continuous or discontinuous
-      transfer.fine_element_is_continuous = fe_fine.n_dofs_per_vertex() > 0;
+      transfer.fine_element_is_continuous  = fe_fine.n_dofs_per_vertex() > 0;
+      unsigned int current_refinement_case = numbers::invalid_unsigned_int;
```

Comment:
```suggestion
      std::uint8_t current_refinement_case = static_cast<std::uint8_t>(-1);
```
and similarly in a few places below.